lib/oplog/tail.go

```diff
@@ -143,56 +172,137 @@ func (tailer *Tailer) tailOnce(out chan<- *redispub.Publication, stop <-chan boo
 
 		var rawData bson.Raw
 
-		for iter.Next(&rawData) {
-			ts, pubs := tailer.unmarshalEntry(rawData)
+		for {
+			gotResult, didTimeout, didLosePosition, err := readNextFromCursor(query)
```

Comment:
i think any more than 4 values returned from a function, time to invent a data type to hold the result

Reply:
Yep, I agree - refactored this a bit and added a comment ..